### ISSUE TITLE
"State" callbacks for pending/reading and idle UI changes

### ISSUE THREAD

**Issue by JamoCA** [CarlRaymond/jquery.cardswipe#4]
During the scanning process, there's no visual indication that anything is happening until the script has completed reading and initiates parsing.  Using Firefox with dev tools, there's a 1-3 second pause during the "pending/reading" process and the operator may be tempted to rescan or press a key.  (Google Chrome is much faster, but I still want to show something on the screen that something is happening.)

I initially added callbacks so that I could block the UI (using BlockUI and "bindEvents:false" option so keyboard would continue to work) and displayed a "Processing..." message or spinner.  Since the "prefix" update, my hacks aren't working properly and I was wondering if you could add these features so I can replace my library with yours without having to use an internal DIFF tool. (I'm not that savvy with GitHub yet.)

I added "onScanStart" & "onScanEnd" options to the config and defaulted both to null.  Here's my attempt (and it's currently 4 commits behind yours.):
https://github.com/JamoCA/jquery.cardswipe

**Comment by rogerskk** [comment]
Any update on this? I would like to have a "Please wait.." show while the card is processing after a scan because of the delay. Thank you.

**Comment by CarlRaymond** [comment]
Actually, yes.  There's a branch called "events" with a preliminary version.  It's preliminary, but I'd like some feedback.  The demo.html page shows how to invoke it, using the built-in parsers.
 
Unlike before, there are multiple parsers, each handling one kind of card. You pass in an array with one element for each kind of parser you want to use, including your own custom one if that's what you need. The build-in ones are referenced by name (a string), and your own would get passed in as a function reference.

Good luck!

**Comment by rogerskk** [comment]
Super. I didn't think to check out the branches.  I will try it out and let you know how it goes. Thanks!

**Comment by rogerskk** [comment]
So, I've been playing around with this today.  Your events sample works great.  I'm trying to use this under Asp.Net MVC and it gives quite a delay and then brings up my please wait spinner and then continues.  Not sure why the long delay before firing that scanstart event.  Any ideas?

**Comment by CarlRaymond** [comment]
Hmm. No idea. I use the same platform, and in any case, the server side shouldn't matter at all.

**Comment by rogerskk** [comment]
It might have to do with the version of jQuery I'm using.  I'm using 2.1.3 and jquery-ui version 1.11.3.  Have you tried it out with those versions?

**Comment by rogerskk** [comment]
Just to update this for anyone who is also having long delays when scanning -- I determined the problem.  I get a long delay when the scanning is inside a @using (Html.BeginForm("PostPayment", "Home", FormMethod.Post, new { id = "submitForm" })){  with a lot of fields on the page.
If I remove the using (Html.BeginForm... then the scan happens very fast.  I think it has something to do with client side validation but I'm not sure.  I tried disabling client validation, but it did not change anything.  I thought perhaps it was going through trying to validate the hundreds of fields I have when I initialize a scan.  So, the problem is not the modal, but the form.
